components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```
@@ -0,0 +1,532 @@
+        Assert.assertEquals(creatingAction.getName(), action.getName());
+        Assert.assertNull(null, action.getDescription());
+        Assert.assertEquals(PRE_ISSUE_ACCESS_TOKEN, action.getType().getActionType());
+        Assert.assertEquals(Action.Status.ACTIVE, action.getStatus());
+        Assert.assertEquals(creatingAction.getEndpoint().getUri(), action.getEndpoint().getUri());
+        Assert.assertEquals(creatingAction.getEndpoint().getAuthentication().getType(),
+                action.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 7)
+    public void testUpdateAction() throws ActionMgtException {
+
+        Action updatingAction = buildMockAction(
+                "Pre Issue Access Token",
+                "To configure pre issue access token",
+                "https://sample.com",
+                 AuthType.AuthenticationType.BASIC,
+                 buildMockBasicAuthProperties("updatingadmin", "updatingadmin"));
+        Action result = daoImpl.updateAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), updatingAction, action, TENANT_ID);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(updatingAction.getName(), result.getName());
+        Assert.assertEquals(updatingAction.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(updatingAction.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(
+                updatingAction.getEndpoint().getAuthentication().getType(),
+                result.getEndpoint().getAuthentication().getType()
+        );
+        action = result;
+    }
+
+    @Test(priority = 8)
+    public void testUpdateActionWithoutNameAndDescription() throws ActionMgtException {
+
+        // TODO: This should be modified once the dao layer is fixed to handle null values.
```

This comment is not right.
The todo should be as below.

'Name' is a required attribute. Thus, DAO layer should throw an exception if name is null. This should be fixed in DAO layer and test case needs to be updated accordingly.

---

Fix the later comments similarly

---

Updated in 2d39577263be9c9d0d132b7d1243aea17b118c8b